crates/oxc_ecmascript/src/to_string.rs

```diff
@@ -70,14 +70,16 @@ impl<'a> ToJsString<'a> for TemplateLiteral<'a> {
 }
 
 impl<'a> ToJsString<'a> for IdentifierReference<'a> {
-    fn to_js_string(&self) -> Option<Cow<'a, str>> {
+    fn to_js_string(&self, is_global_reference: &impl IsGlobalReference) -> Option<Cow<'a, str>> {
         let name = self.name.as_str();
-        matches!(name, "undefined" | "Infinity" | "NaN").then(|| Cow::Borrowed(name))
+        (matches!(name, "undefined" | "Infinity" | "NaN")
+            && is_global_reference.is_global_reference(self) == Some(true))
+        .then_some(Cow::Borrowed(name))
     }
```

Comment:
The main change is this part. The other parts are just passing down `IsGlobalReference`.